src/tx/builder/index.ts

```diff
@@ -0,0 +1,743 @@
+   * @default 'max'
+   *
+   */
+  strategy?: 'max' | 'continuity'
+
+  /**
+   * Should be set if you want the transaction
+   * to be only valid until a certain block height is reached.
+   * @default 0
+   */
+  ttl?: number
+
+  /**
+   * The minimum fee is dependent on the tx-type.
+   *
+   * The default value is calculated for each tx-type.
+   */
+  fee?: number
+
+  /**
+   * Should be used for signing an inner transaction that will be wrapped in a `PayingForTx`.
+   * @default false
+   */
+  innerTx?: boolean
+
+  /**
+   * If set to true the transaction will be verified prior to broadcasting it.
+   * @default false
+   */
+  verify?: boolean
+
+  /**
+   * Wait for transactions to be mined.
+   *
+   * You can get the tx object that contains the tx-hash immediately by setting to `false`
+   * and should implement your own logic to watch for mined transactions.
+   *
+   * @default true
+   */
+  waitMined?: boolean
+}
+interface TxOptionsContract extends TxOptionsCommon {
+  /**
+   * To be used for providing `aettos` (or `AE` with respective denomination)
+   * to a contract related transaction.
+   * @default 0
+   */
+  amount?: number
+
+  /**
+   * You can specify the denomination of the `amount`
+   * that will be provided to the contract related transaction.
+   * @default 'aettos'
+   */
+  denomination?: string
+
+  /**
+   * Maximum amount of gas to be consumed by the transaction.
+   * Learn more on {@link https://github.com/aeternity/aepp-sdk-js/blob/develop/docs/transaction-options.md#how-to-estimate-gas How to estimate gas?}
+   */
+  gasLimit?: number
+
+  /**
+   * To increase chances to get your transaction included quickly you can use a higher gasPrice.
+   * @default 1e9
+   */
+  gasPrice?: number
+}
+
+interface TxOptionsNameUpdate extends TxOptionsCommon {
+  /**
+   * This option is an indicator for indexing tools to know how long (in seconds)
+   * they could or should cache the name information.
+   * @default 84600
+   */
+  clientTtl?: number
+
+  /**
+    * This option tells the protocol the relative TTL based on the current block height.
+    * @default 180000
+    */
+  nameTtl?: number
+}
+
+interface TxOptionsNameClaim extends TxOptionsCommon {
+  /**
+   * The fee in `aettos` that will be payed to claim the name.
+   *
+   * For bids in an auction you need to explicitely calculate the required `nameFee`
+   * based on the last bid
+   *
+   * The default value is calculated based on the length of the name
+   */
+  nameFee?: number
+}
+
+interface TxOptionsOracleRegister extends TxOptionsCommon {
+  /**
+   * The fee in `aettos` that the oracle requests in order to provide a response.
+   * @default 30000
+   */
+  queryFee?: number
+
+  /**
+   * The TTL of the oracle that defines its expiration.
+   * @default {type:'delta', value:500}
+   */
+  oracleTtl?: {
+    type?: 'delta' | 'block'
+    value?: number
+  }
+}
+
+interface TxOptionsOracleQuery extends TxOptionsCommon {
```

Comment:
The interfaces like this one I assume should be generated by schema (see my example in the last commit)